### PR TITLE
[OCPBUGS#35239]: Update the compatibility matrix

### PR DESCRIPTION
**Pull request by tmalove**
Jira: [OCPBUGS-35239](https://issues.redhat.com/browse/OCPBUGS-35239)

Version(s):
This PR is based on the ‘opp-docs-main’ repo and when merged, it should be in that branch only. The OPP doc is not versioned. Add this PR to the ‘Continuous Release’ milestone.

Link to docs preview:
[Architecture](https://77314--ocpdocs-pr.netlify.app/openshift-opp/latest/architecture/opp-architecture.html) Updated 6/26/2024)

QE review:
- [x ] QE has approved this change. (Oded R.)

Additional information:
The product versions that are listed might not be the latest available version for the product. The versions listed in the compatibility matrix are the latest verified versions for OPP.

### CODE REVIEW

**Comment by tmalove** [comment]
@gparvin  Will you take a look at the compatibility matrix ?  I added the support product versions back to 4.12, but not sure if these versions were actually verified based on the OCP version. I will send Oded a DM since I could not find his @ in GH. Thanks!

Also, the architectural update diagram updates are coming soon.

File: _attributes/common-attributes.adoc [hunk 1]
```diff
@@ -37,11 +37,33 @@
...
 :olm-first: Operator Lifecycle Manager (OLM)
 :olm: OLM
-:ocp-supported-version: 4.15
-:rhacs-version: 4.4
-:quay-version: 3.11
+// These variable are the current product release versions for the OPP products
+:ocp-supported-version: 4.16
+:rhacs-version: 4.5
+:quay-version: 3.12
 :rhacm-version: 2.10
```
Comment (gparvin) [inline _attributes/common-attributes.adoc:44]: We will have an ACM 2.11 release that will be available not long after OCP 4.16 is available.  That is the one that should be listed here.

File: _attributes/common-attributes.adoc [hunk 2]
```diff
@@ -37,11 +37,33 @@
...
+:odf-version-1: 4.15
+:ocp-supported-version-2: 4.14
+:rhacs-version-2: 4.3
+:quay-version-2: 3.10
+:rhacm-version-2: 2.9
+:odf-version-2: 4.14
+:ocp-supported-version-3: 4.13
+:rhacs-version-3: 4.0
+:quay-version-3: 3.8
+:rhacm-version-3: 2.7
```
Comment (gparvin) [inline _attributes/common-attributes.adoc:60]: We should have had ACM 2.8 here.  Can you double check to see when we documented ACM 2.8?

Comment (tmalove) [reply _attributes/common-attributes.adoc]: @gparvin after looking at the previous PRs to update the versions, 2.8 follows the trend, however, I will have Oded verify this. Thank you!

**Comment by tmalove** [comment]
@gparvin  I modified the original scope of this PR  from Updating the architecture overview document and modifying the compatibility matrix to just the matrix updates. I will open a PR for updating the diagram when those changes are underway.
In the meantime, will you ack the matrix changes? You and Oded agreed that ACM 2.8 was verified with OCP 4.13. Thanks!

**Review by gparvin — APPROVED** [review]
This looks right to me.

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

**Review by opayne1 — COMMENTED** [review]
/lgtm

**Pull request merged**